### ISSUE TITLE
SingleThreadScheduler throws a "RejectedExecutionException: Thread pool shutdown" when it has not been shutdown

### ISSUE THREAD

**Issue by lwahlmeier** [threadly/threadly#126]
Here is a simple unit test to reproduce the issue:

```java
  @Test
  public void stsTest() {
    final int number = 500000;
    PriorityScheduler CLIENT_PS = new PriorityScheduler(5, 5, 100000);
    final SingleThreadScheduler SSI = new SingleThreadScheduler();
    //final PriorityScheduler SSI = new PriorityScheduler(1, 1, 100000);
    final AtomicInteger count = new AtomicInteger(0);
    Runnable run = new Runnable() {
      @Override
      public void run() {
        SSI.execute(new Runnable() {

          @Override
          public void run() {
            count.incrementAndGet();
          }});
      }
    };
    for(int i=0; i<number; i++) {
      CLIENT_PS.execute(run);
    }
    
    new TestCondition(){
      @Override
      public boolean get() {
        System.out.println(count.get());
        return count.get() == number;
      }
    }.blockTillTrue(10000, 1000);
  }
```

when ran you will see:
Exception in thread "PriorityScheduler-1-thread-6" java.util.concurrent.RejectedExecutionException: Thread pool shutdown
	at org.threadly.concurrent.SingleThreadScheduler.getRunningScheduler(SingleThreadScheduler.java:93)
	at org.threadly.concurrent.SingleThreadScheduler.doSchedule(SingleThreadScheduler.java:222)
	at org.threadly.concurrent.AbstractSubmitterScheduler.doExecute(AbstractSubmitterScheduler.java:23)
	at org.threadly.concurrent.AbstractSubmitterExecutor.execute(AbstractSubmitterExecutor.java:33)
	at org.threadly.litesockets.protocol.http.HTTPClientTests$1.run(HTTPClientTests.java:62)
	at org.threadly.concurrent.PriorityScheduler$OneTimeTaskWrapper.run(PriorityScheduler.java:1509)
	at org.threadly.util.ExceptionUtils.runRunnable(ExceptionUtils.java:91)
	at org.threadly.concurrent.PriorityScheduler$Worker.run(PriorityScheduler.java:1407)
	at java.lang.Thread.run(Thread.java:745)

if you comment out the SingleThreadScheduler and use a singleThreaded PriorityScheduler this is not an issue.

**Comment by jentfoo** [comment]
@lwahlmeier I pasted the unit test you provided exactly as you have it and I am not able to reproduce the issue.

One thought I had...are you shutting down the scheduler in an "@After" function, could this exception be from a test which has finished running?

**Comment by jentfoo** [comment]
@lwahlmeier second thought...do you have this test in a branch on the litesockets repo?  If so, I don't mind taking a look at everything you have and help nail it down.

**Comment by lwahlmeier** [comment]
I definitely dont have any @Before or @After in the test.  I made a whole new test class just for this issue with only this test in it.  I just moved to the latest master and can still reproduce it.  It takes about 10 trys on my laptop and sometimes(very rarely) it does not throw and just hard loops.  

I am pretty sure it has to do with doing .executes from different threads when the thread has not been created.

if I add:
```java
SSI.submit(new Runnable() {
      @Override
      public void run() {
      }}).get();
```
After the creation of the SingleThreadScheduler I stop being able to reproduce it.  

I also think the more core the computer running the test has the more likely it is to reproduce, and I am fairly certain if ran on a single core system it would not reproduce at all.  If you need to add more then 5 threads to the CLIENT_PS, assuming you have more cores for it.

**Comment by jentfoo** [comment]
Good spot, I think I know what's going on...I will give this a deeper look

**Comment by jentfoo** [comment]
Fixed with pull request #128
